pydantic_ai_slim/pydantic_ai/models/openai.py

```diff
@@ -1252,6 +1254,8 @@ async def _responses_create(
             if (status_code := e.status_code) >= 400:
                 raise ModelHTTPError(status_code=status_code, model_name=self.model_name, body=e.body) from e
             raise  # pragma: lax no cover
+        except APIConnectionError as e:
+            raise ModelHTTPError(status_code=0, model_name=self.model_name, body=str(e)) from e
```

Comment:
As I'm sure you also noticed when you wrote this, we're kind of abusing the `ModelHTTPError` class now because this is not actually an HTTP response with status code 0.

I suggest we add a new superclass of `ModelHTTPError` like `ModelAPIError` without a `status_code` field, and raise that instead.

We should update `FallbackModel` to have that in the default `on_fallback` list (and update the docs if needed)

And I think we should review every single model class and raise the new error for non-HTTP errors.

I think Bedrock currently incorrectly raises `ModelHTTPError` for non-HTTP errors as well, so that one will need further tweaking.